examples/7guis/cells.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+use slint::{Model, ModelRc, SharedString};
+use std::cell::RefCell;
+use std::collections::{HashMap, HashSet};
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::rc::{Rc, Weak};
+
+slint::slint!(import { MainWindow } from "cells.slint";);
+
+const ROW_COUNT: usize = 100;
+const COL_COUNT: usize = 26;
+
+/// Graph of dependences so that node B depends on node A.
+/// So when node A is updated, node B also need to re updated.
+#[derive(Default)]
+struct DepGraph<A, B> {
+    index1: HashMap<A, HashSet<B>>,
+    index2: HashMap<B, HashSet<A>>,
+}
+
+impl<A: Clone + Eq + Hash, B: Clone + Eq + Hash> DepGraph<A, B> {
+    pub fn add_dep(&mut self, a: A, b: B) {
+        self.index1.entry(a.clone()).or_default().insert(b.clone());
+        self.index2.entry(b).or_default().insert(a);
+    }
+
+    pub fn deps<'a>(&'a self, a: &A) -> impl Iterator<Item = &B> + 'a {
+        self.index1.get(a).into_iter().flat_map(|x| x.iter())
+    }
+
+    pub fn remove_rev_deps(&mut self, b: &B) {
+        if let Some(h) = self.index2.remove(b) {
+            for a in h {
+                self.index1.get_mut(&a).map(|x| x.remove(b));
+            }
+        }
+    }
+}
+
+#[derive(Debug, PartialEq)]
+enum Expr {
+    Value(f32),
+    Cell(usize, usize),
```

Comment:
Small comment to make it easier to read :)

```suggestion
    Cell(usize, usize), // (row, column)
```